include/freerdp/server/rail.h

```diff
@@ -0,0 +1,154 @@
+typedef UINT(*psRailServerTaskbarInfo)(RailServerContext* context,
+                                       const RAIL_TASKBAR_INFO_ORDER* taskbarInfo);
+typedef UINT(*psRailServerLangbarInfo)(RailServerContext* context,
+                                       const RAIL_LANGBAR_INFO_ORDER* langbarInfo);
+typedef UINT(*psRailServerExecResult)(RailServerContext* context,
+                                      const RAIL_EXEC_RESULT_ORDER* execResult);
+typedef UINT(*psRailServerGetAppIdResp)(RailServerContext* context,
+                                        const RAIL_GET_APPID_RESP_ORDER* getAppIdResp);
+typedef UINT(*psRailServerZOrderSync)(RailServerContext* context,
+                                      const RAIL_ZORDER_SYNC_ORDER* zOrderSync);
+typedef UINT(*psRailServerCloak)(RailServerContext* context,
+                                 const RAIL_CLOAK_ORDER* cloak);
+typedef UINT(*psRailServerPowerDisplayRequest)(RailServerContext* context,
+        const RAIL_POWER_DISPLAY_REQUEST_ORDER* PowerDisplayRequest);
+typedef UINT(*psRailServerGetAppidRespEx)(RailServerContext* context,
+        const RAIL_GET_APPID_RESP_EX_ORDER* GetAppidRespEx);
+
+
+struct _rail_server_context
+{
+	HANDLE vcm;
+	void* custom;
+
+	psRailStart Start;
+	psRailStop Stop;
+
+	/* Callbacks from client */
+	psRailClientHandshake ClientHandshake;
+	psRailClientClientStatus ClientClientStatus;
+	psRailClientExec ClientExec;
+	psRailClientSysparam ClientSysparam;
+	psRailClientActivate ClientActivate;
+	psRailClientSysmenu ClientSysmenu;
+	psRailClientSyscommand ClientSyscommand;
+	psRailClientNotifyEvent ClientNotifyEvent;
+	psRailClientGetAppidReq ClientGetAppidReq;
+	psRailClientWindowMove ClientWindowMove;
+	psRailClientSnapArrange ClientSnapArrange;
+	psRailClientLangbarInfo ClientLangbarInfo;
+	psRailClientLanguageImeInfo ClientLanguageImeInfo;
+	psRailClientCompartmentInfo ClientCompartmentInfo;
+	psRailClientCloak ClientCloak;
+
+	/* Methods for sending server side messages */
+	psRailServerHandshake ServerHandshake;
+	psRailServerHandshakeEx ServerHandshakeEx;
+	psRailServerSysparam ServerSysparam;
+	psRailServerLocalMoveSize ServerLocalMoveSize;
+	psRailServerMinMaxInfo ServerMinMaxInfo;
+	psRailServerTaskbarInfo ServerTaskbarInfo;
+	psRailServerLangbarInfo ServerLangbarInfo;
+	psRailServerExecResult ServerExecResult;
+	psRailServerGetAppIdResp ServerGetAppIdResp;
+	psRailServerZOrderSync ServerZOrderSync;
+	psRailServerCloak ServerCloak;
+	psRailServerPowerDisplayRequest ServerPowerDisplayRequest;
+	psRailServerGetAppidRespEx ServerGetAppidRespEx;
+
+	RailServerPrivate* priv;
+	rdpContext* rdpcontext;
+};
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+FREERDP_API RailServerContext* rail_server_context_new(HANDLE vcm);
+FREERDP_API void rail_server_context_free(RailServerContext* context);
+FREERDP_API HANDLE rail_server_get_event_handle(RailServerContext* context);
+FREERDP_API UINT rail_server_handle_messages(RailServerContext* context);
```

Comment:
why its exported?

Reply:
I'm not done with the API yet, but this is what they export in every other channel